2006-01-04	Arnaud Charlet	<charlet@adacore.com>

	* gtk/gtktextview.c (gtk_text_view_start_selection_dnd): Disabled,
	since not functional.
	For C928-003.

2003-11-27	Arnaud Charlet	<charlet@adacore.com>

	* gtk/gtktextview.c (incremental_validate_callback): Increase amount
	of characters validated at each call, this gives a better behavior
	when loading big files.
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=332057

	* gtk/gtktextbtree.c, gtk/gtktextiter.c: Remove some function calls
	when assertions are disabled.
	Submitted at http://bugzilla.gnome.org/show_bug.cgi?id=332059

--- gtk/gtktextview.c.old	2006-02-17 16:11:04.000000000 +0100
+++ gtk/gtktextview.c	2006-02-20 17:08:54.691581240 +0100
@@ -3273,7 +3273,7 @@ incremental_validate_callback (gpointer 
   
   DV(g_print(G_STRLOC"\n"));
   
-  gtk_text_layout_validate (text_view->layout, 2000);
+  gtk_text_layout_validate (text_view->layout, 20000);
 
   gtk_text_view_update_adjustments (text_view);
   
@@ -6028,6 +6028,8 @@ gtk_text_view_start_selection_dnd (GtkTe
   GdkDragContext *context;
   GtkTargetList  *target_list;
 
+  return;
+
   text_view->drag_start_x = -1;
   text_view->drag_start_y = -1;
   text_view->pending_place_cursor_button = 0;
--- gtk/gtktextbtree.c.old	2006-02-17 16:11:31.000000000 +0100
+++ gtk/gtktextbtree.c	2006-02-17 16:12:02.000000000 +0100
@@ -3348,13 +3348,9 @@ ensure_end_iter_line (GtkTextBTree *tree
 {
   if (tree->end_iter_line_stamp != tree->chars_changed_stamp)
     {
-      int n_lines;
       int real_line;
 
-      /* n_lines is without the magic line at the end */
-      n_lines = _gtk_text_btree_line_count (tree);
- 
-      g_assert (n_lines >= 1);
+      g_assert (_gtk_text_btree_line_count (tree) >= 1);
 
       tree->end_iter_line = _gtk_text_btree_get_line_no_last (tree, -1, &real_line);
       
--- gtk/gtktextiter.c.old	2006-02-17 16:12:09.000000000 +0100
+++ gtk/gtktextiter.c	2006-02-17 16:16:50.000000000 +0100
@@ -152,6 +152,11 @@ iter_set_from_segment (GtkTextRealIter *
    truly computed lazily; often we don't need to do the full make_real
    work. This ensures the btree and line are valid, but doesn't
    update the segments. */
+
+#ifdef G_DISABLE_ASSERT
+#define gtk_text_iter_make_surreal(iter) ((GtkTextRealIter*)iter)
+
+#else
 static GtkTextRealIter*
 gtk_text_iter_make_surreal (const GtkTextIter *_iter)
 {
@@ -190,6 +195,7 @@ gtk_text_iter_make_surreal (const GtkTex
 
   return iter;
 }
+#endif
 
 static GtkTextRealIter*
 gtk_text_iter_make_real (const GtkTextIter *_iter)
@@ -387,15 +393,15 @@ is_segment_start (GtkTextRealIter *real)
   return real->segment_byte_offset == 0 || real->segment_char_offset == 0;
 }
 
-#if 1
+#ifdef G_DISABLE_ASSERT
+#define check_invariants(iter)
+#else
 static void
 check_invariants (const GtkTextIter *iter)
 {
   if (gtk_debug_flags & GTK_DEBUG_TEXT)
     _gtk_text_iter_check (iter);
 }
-#else
-#define check_invariants (x)
 #endif
 
 /**
